pkg/providers/instance/instance.go

```diff
@@ -522,7 +528,19 @@ func (p *DefaultProvider) launchInstance(
 	sshPublicKey := options.FromContext(ctx).SSHPublicKey
 	nodeIdentityIDs := options.FromContext(ctx).NodeIdentities
 	useSIG := options.FromContext(ctx).UseSIG
-	vm := newVMObject(resourceName, nicReference, zone, capacityType, p.location, sshPublicKey, nodeIdentityIDs, nodeClass, launchTemplate, instanceType, p.provisionMode, useSIG)
+	vm := newVMObject(
```

Comment:
This is maybe excessive - possibly could revert.

Really this function has too many args.

Reply:
It does feel a little excessive

Reply:
I'll revert this, I don't love the formatting of it either.

Probably we should revisit this function and reduce its args at some point, but I am not going to do that here.